catalog/core/catalog-core-standardframework/src/test/java/org/codice/ddf/catalog/resource/cache/impl/ResourceCacheServiceTest.java

```diff
@@ -0,0 +1,215 @@
+    }
+
+    @Test
+    public void testContainsResourceInCache() throws Exception {
+        // Setup
+        setupMockDownloadManager(true);
+        setupMockResourceCache(true);
+
+        // Perform Test
+        ResourceCacheServiceMBean resourceCacheService = createResourceCacheServiceMBean();
+        boolean isMetacardCached = resourceCacheService.contains(metacard);
+
+        assertThat(isMetacardCached, is(true));
+    }
+
+    @Test
+    public void testInit() throws Exception {
+        // Setup
+        setupMockMBeanServer(false, false);
+        ResourceCacheService resourceCacheService = (ResourceCacheService) createResourceCacheServiceMBean();
+
+        // Perform Test
+        resourceCacheService.init();
+
+        verify(mockMBeanServer).registerMBean(any(StandardMBean.class),
+                eq(resourceCacheServiceObjectName));
+    }
+
+    @Test
+    public void testInitMBeanAlreadyRegistered() throws Exception {
+        // Setup
+        setupMockMBeanServer(true, false);
+        ResourceCacheService resourceCacheService = (ResourceCacheService) createResourceCacheServiceMBean();
+
+        // Perform Test
+        resourceCacheService.init();
+
+        verify(mockMBeanServer, never()).registerMBean(any(StandardMBean.class),
+                eq(resourceCacheServiceObjectName));
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testInitMBeanMBeanRegistrationFails() throws Exception {
+        // Setup
+        setupMockMBeanServer(false, true);
+        ResourceCacheService resourceCacheService = (ResourceCacheService) createResourceCacheServiceMBean();
+
+        // Perform Test
+        resourceCacheService.init();
+    }
+
+    @Test
+    public void testDestroy() throws Exception {
+        // Setup
+        setupMockMBeanServer(true, false);
+        ResourceCacheService resourceCacheService = (ResourceCacheService) createResourceCacheServiceMBean();
+
+        // Perform Test
+        resourceCacheService.destroy();
+
+        verify(mockMBeanServer).unregisterMBean(eq(resourceCacheServiceObjectName));
+    }
+
+    @Test
+    public void testDestroyMBeanNotRegistered() throws Exception {
+        // Setup
+        setupMockMBeanServer(false, false);
+        ResourceCacheService resourceCacheService = (ResourceCacheService) createResourceCacheServiceMBean();
+
+        // Perform Test
+        resourceCacheService.destroy();
+
+        verify(mockMBeanServer, never()).unregisterMBean(eq(resourceCacheServiceObjectName));
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testDestroyMBeanUnregistrationFails() throws Exception {
+        // Setup
+        setupMockMBeanServer(true, true);
+        ResourceCacheService resourceCacheService = (ResourceCacheService) createResourceCacheServiceMBean();
+
+        // Perform Test
+        resourceCacheService.destroy();
+    }
+
+    private void setupMockMBeanServer(boolean isRegistered, boolean throwMBeanRegException)
```

Comment:
We probably should have a couple tests (success and exception) for the `get(Metacard)` method.